songpal/device.py

```diff
@@ -232,7 +278,26 @@ async def get_interface_information(self) -> InterfaceInfo:
 
     async def get_system_info(self) -> Sysinfo:
         """Return system information including mac addresses and current version."""
-        return Sysinfo.make(**await self.services["system"]["getSystemInformation"]())
+
+        if self.services["system"].has_method("getSystemInformation"):
+            return Sysinfo.make(
+                **await self.services["system"]["getSystemInformation"]()
+            )
+        elif self.services["system"].has_method("getNetworkSettings"):
+            info = await self.services["system"]["getNetworkSettings"](netif="")
+
+            def get_addr(info, iface):
+                addr = next((i for i in info if i["netif"] == iface), {}).get("hwAddr")
+                return addr.lower().replace("-", ":") if addr else addr
+
+            macAddr = get_addr(info, "eth0")
+            wirelessMacAddr = get_addr(info, "wlan0")
```

Comment:
Response from `getNetworkSettings` should be contained in its own container, `NetworkSettings` maybe?
Here's what the structure contains based on some logs I have:
> DEBUG:songpal.method:system.getNetworkSettings outs: {'netif': <class 'str'>, 'hwAddr': <class 'str'>, 'ipAddrV4': <class 'str'>, 'ipAddrV6': <class 'str'>, 'netmask': <class 'str'>, 'gateway': <class 'str'>, 'dns': 'string*'}